packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts

```diff
@@ -240,19 +249,83 @@ export abstract class BaseExportPodCommand<
         if (!selection) {
           return resolve(undefined);
         }
-        const { hierarchy, vault } = selection
-        const notes = ExtensionProvider.getEngine().notes;
+        const { hierarchy, vault } = selection;
+        const notes = this.extension.getEngine().notes;
 
         resolve(
           Object.values(notes).filter(
-            (value) => value.fname.startsWith(hierarchy) && value.stub !== true &&
-             VaultUtils.isEqualV2(value.vault, vault)
+            (value) =>
+              value.fname.startsWith(hierarchy) &&
+              value.stub !== true &&
+              VaultUtils.isEqualV2(value.vault, vault)
           )
         );
       });
     });
   }
 
+  /**
+   * If the active text editor document has dirty changes, save first before exporting
+   * @returns True if document is dirty, false otherwise
+   */
+  private async saveActiveDocumentBeforeExporting(opts: {
+    config: Config;
+    payload: NoteProps[];
+  }): Promise<boolean> {
+    const editor = VSCodeUtils.getActiveTextEditor();
+    if (editor && editor.document.isDirty) {
+      const fname = NoteUtils.uri2Fname(editor.document.uri);
+      this._onEngineNoteStateChangedDisposable = this.extension
+        .getEngine()
+        .getEngineEmitter()
+        .onEngineNoteStateChanged(
+          async (noteChangeEntries: NoteChangeEntry[]) => {
+            const updateNoteEntries = noteChangeEntries.filter(
+              (entry) => entry.note.fname === fname && entry.status === "update"
+            );
+            // Received event from engine about successful save
+            if (updateNoteEntries.length > 0) {
+              const savedNote = updateNoteEntries[0].note;
+              // Remove notes that match saved note as they contain old content
+              const filteredPayload = opts.payload.filter(
+                (note) => note.fname !== savedNote.fname
+              );
+              await this.executeExportNotes({
+                ...opts,
+                payload: filteredPayload.concat(savedNote),
+              });
+              this.dispose();
+            }
+          }
+        );
+      await editor.document.save();
+      // Dispose of listener after 1 sec (if not already disposed) in case engine events never arrive
+      setTimeout(() => {
+        this.dispose();
```

Comment:
we should notify the user of an error in this case and tell them that their export didn't happen.

Reply:
done